### PR TITLE
Update Wasmtime for upcoming WIT changes

### PR DESCRIPTION
**Pull request by alexcrichton**
This PR integrates bytecodealliance/wasm-tools#1027 into Wasmtime. The main changes here are:

* WIT syntax is updated with WebAssembly/component-model#193
* Generated bindings in the `bindgen!` macro have been updated to reflect the new structure of WIT.
* The accepted component model binary format has been updated to account for changes.

This PR disables wasi-http tests and the on-by-default feature because the WIT syntax has been updated but the submodule containing the WITs has not been updated yet so there's no way to get that building temporarily. Once that's updated then this can be reenabled.

### CODE REVIEW

**Comment by alexcrichton** [comment]
I'll note that I'm opening this as a draft as I'm working my way up the stack. My goal is to prove out https://github.com/bytecodealliance/wasm-tools/pull/1027 and once everything is ready have a day I spend landing the various changes.

**Comment by alexcrichton** [comment]
Ok I've now updated this for the final set of merged changes to wasm-tools, which aren't yet published. The major changes in this PR are now:

* All wasm-tools deps are updated (currently sitting at git deps to my "bump" branch)
* WIT syntax is changing, as well as the binary format of components
* All WIT files vendored in this repository are updated to the new syntax. Notably the wasi-http WITs are not updated because they're a submodule. This means that all wasi-http-related tests and support are disabled.
* There is now only one copy of WIT for WASI, and all other duplicate copies are deleted.
* All of the preview2 implementation has been updated for the new hierarchy of modules created by `wasmtime::component::bindgen!`

One interesting aspect of this is that there's a temporary git dependency on wit-bindgen. I'm not actually sure how to get rid of that because the adapter requires wit-bindgen and it's not easy to remove that. This is somewhat ok though since we don't publish anything with wit-bindgen so I'm hoping it's ok to have a temporary state of affairs with a git dep here before I can go use this PR to update wit-bindgen to make a release there.

**Comment by alexcrichton** [comment]
cc @rvolosatovs and @sunfishcode as y'all are likely interested in the various changes here too

File: crates/wasi/wit/main.wit [hunk 1]
```diff
@@ -0,0 +1 @@
+package unused:main
```
Comment (sunfishcode) [inline crates/wasi/wit/main.wit:1]: What is this `main.wit` for?

**Review by pchickey — APPROVED** [review]

**Comment by alexcrichton** [comment]
Can you confirm that `wit-component` 0.11.0 was used? The version `0xc` is the "old" version of components and the upgraded versions should be `0xd`, so it may be the case that `wit-component 0.11.0` wasn't used by accident?

**Comment by rvolosatovs** [comment]
> Can you confirm that `wit-component` 0.11.0 was used? The version `0xc` is the "old" version of components and the upgraded versions should be `0xd`, so it may be the case that `wit-component 0.11.0` wasn't used by accident?

I'll try to minimize this later today

**Pull request merged**